.eslintrc.json

```diff
@@ -48,45 +46,36 @@
             { "selector": "property", "format": null }
         ],
 
-        "@typescript-eslint/consistent-type-definitions": ["error", "interface"],
-        "@typescript-eslint/consistent-type-assertions": ["error", { "assertionStyle": "as" }],
-
-        "max-statements-per-line": ["error", { "max": 1 }],
-
-        "no-duplicate-imports": "off",
-        "@typescript-eslint/no-duplicate-imports": "error",
-
-        "@typescript-eslint/no-inferrable-types": "error",
-        "@typescript-eslint/no-misused-new": "error",
-        "@typescript-eslint/no-this-alias": "error",
-
-        "no-unused-expressions": "off",
-        "@typescript-eslint/no-unused-expressions": ["error", { "allowTernary": true }],
-
-        "@typescript-eslint/prefer-for-of": "error",
-        "@typescript-eslint/prefer-function-type": "error",
-        "@typescript-eslint/prefer-namespace-keyword": "error",
-        "@typescript-eslint/prefer-as-const": "error",
-
-        "quotes": "off",
+        "@typescript-eslint/brace-style": ["error", "stroustrup", { "allowSingleLine": true }],
+        "@typescript-eslint/no-extra-semi": "error",
         "@typescript-eslint/quotes": ["error", "double", { "avoidEscape": true, "allowTemplateLiterals": true }],
-
-        "semi": "off",
         "@typescript-eslint/semi": "error",
-        "@typescript-eslint/no-extra-semi": "error",
-
-        "space-before-function-paren": "off",
         "@typescript-eslint/space-before-function-paren": ["error", {
             "asyncArrow": "always",
             "anonymous": "always",
             "named": "never"
         }],
 
-        "@typescript-eslint/triple-slash-reference": "error",
-        "@typescript-eslint/type-annotation-spacing": "error",
-        "@typescript-eslint/unified-signatures": "error",
-
-        "@typescript-eslint/no-extra-non-null-assertion": "error",
+        // Todo: do we want these?
+        "prefer-rest-params": "off",
```

Comment:
In the abstract, I'd prefer this to be on (unless there's some horrible efficiency problem). How much needs to change to make that happen?

Reply:
There are 27 errors right now. They're generally pretty straightforward to resolve:

`
``diff
- export function getDiagnosticText(_message: DiagnosticMessage, ..._args: any[]): string {
-    const diagnostic = createCompilerDiagnostic.apply(undefined, arguments);
+ export function getDiagnosticText(message: DiagnosticMessage, ...args: any[]): string {
+    const diagnostic = createCompilerDiagnostic(message, ...args);
```

I'd suggest leaving this for a separate PR just in case one happens to be a surprisingly hot path.

Reply:
There are 14 errors right now. They're mostly pretty straightforward to resolve. E.g.:

```diff
- export function getDiagnosticText(_message: DiagnosticMessage, ..._args: any[]): string {
-    const diagnostic = createCompilerDiagnostic.apply(undefined, arguments);
+ export function getDiagnosticText(message: DiagnosticMessage, ...args: any[]): string {
+    const diagnostic = createCompilerDiagnostic(message, ...args);
```

I'd suggest leaving this for a separate PR to keep things small and just in case one happens to be a surprisingly hot path.

Reply:
We actually have a compiler rule that gets mad at this one too, more or less; it's one I've been trying to figure out how to enable but haven't found an ergonomic way for places like this,